File: .old-compat/src/main/java/androidx/preference/PreferenceScreen.java
package androidx.preference;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

public class PreferenceScreen {
	public List<Preference> preferences = new ArrayList<>();
	
	public boolean addPreference(Preference preference) {
		return preferences.add(preference);
	}
	
	public Context getContext() {
		throw new UnsupportedOperationException();
	}
}

File: extension/loaders/android-compat/src/desktopMain/kotlin/androidx/preference/PreferenceScreen.kt
package androidx.preference

import android.content.Context
import com.mrboomdev.awery.core.utils.PlatformSdk

@PlatformSdk
class PreferenceScreen {
	private val prefs = mutableListOf<Preference>()
	
	fun getPreferenceCount(): Int {
		return prefs.size
	}
	
	fun getPreference(index: Int): Preference {
		return prefs[index]
	}
	
	@PlatformSdk
	fun addPreference(preference: Preference): Boolean {
		return prefs.add(preference)
	}
	
	@PlatformSdk
	fun getContext(): Context {
		throw RuntimeException("Stub!")
	}
}